### ISSUE TITLE
Appium Server Not Picking Up the Non Server Arguments When Started From Config File

### ISSUE THREAD

**Issue by primitivetypo** [appium/appium#17446]
I am trying to add arguments for allow-insecure . It works with CLI but not using the config.yaml.
Here is the command that works if starting appium from CLI
`appium server --allow-insecure=adb_shell,chromedriver_autodownload`

And from appium logs i see correctly the two non default server arguments updated:
```
info Appium Welcome to Appium v2.0.0-beta.21 (REV 3748bed378401ed75abdf32bcb3d2674d854a6f9)
info Appium Non-default server args:
info Appium   allowInsecure: {
info Appium     0: adb_shell
info Appium     1: chromedriver_autodownload
info Appium   }
info Appium Attempting to load driver uiautomator2...
```
However, issue is with starting from config as below:
```
server:
  port: 4723
  keep-alive-timeout: 80
  session-override: true
  allow-insecure:
    - adb_shell
    - chromedriver_autodownload
  use-drivers:
    - uiautomator2

```
When i run
`appium server --config appium.yml` -  I don't see those server arguments activated

Below are the logs:
```
info Appium Welcome to Appium v2.0.0-beta.21 (REV 3748bed378401ed75abdf32bcb3d2674d854a6f9)
info Appium Attempting to load driver uiautomator2...
info Appium Appium REST http interface listener started on 0.0.0.0:4723
info Appium Available drivers:
info Appium   - uiautomator2@2.4.2 (automationName 'UiAutomator2')
info Appium Available plugins:
info Appium   - device-farm@1.0.0-beta.15
info Appium   - appium-dashboard@v1.0.0-beta.12
info Appium No plugins activated. Use the --plugins flag with names of plugins to activate
```
Any idea what i might be missing?

*INFO*
Appium version (or git revision) that exhibits the issue: 2.0 beta.21
Last Appium version that did not exhibit the issue (if applicable):
Desktop OS/version used to run Appium: Monterey
Node.js version (unless using Appium.app|exe): 16.16
Npm or Yarn package manager: 8.11
Mobile platform/version under test: Android
Real device or emulator/simulator: Emulator
Appium CLI or Appium.app|exe: CLI

**Comment by mykola-mokhnach** [comment]
> info Appium Welcome to Appium v2.0.0-beta.21 (REV 3748bed378401ed75abdf32bcb3d2674d854a6f9)

Please update the server to the most recent release

**Comment by primitivetypo** [comment]
Thanks @mykola-mokhnach Issue fixed after upgrading to version v2.0.0-beta.43. Closing as not as issue.
